src/solc_0.7/MultiGiveaway/MultiGiveawayWithERC20.sol

```diff
@@ -0,0 +1,110 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.7.5;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+import "./ClaimERC721AndERC1155WithERC20.sol";
+import "../common/BaseWithStorage/WithAdmin.sol";
+
+/// @title MultiGiveaway contract.
+/// @notice This contract manages multiple ERC721,ERC1155 and ERC20 claims.
+contract MultiGiveawayWithERC20 is WithAdmin, ClaimERC721AndERC1155WithERC20 {
+    bytes4 private constant ERC1155_RECEIVED = 0xf23a6e61;
+    bytes4 private constant ERC1155_BATCH_RECEIVED = 0xbc197c81;
+    bytes4 internal constant ERC721_RECEIVED = 0x150b7a02;
+    bytes4 internal constant ERC721_BATCH_RECEIVED = 0x4b808c46;
+    uint256 internal immutable _expiryTime;
+    mapping(address => bool) public claimed;
+
+    constructor(
+        address asset,
+        address land,
+        address erc20Token,
```

Comment:
Same like above, the erc20 token should be part of the leaf data, so we can give any erc20 tokens